### PR TITLE
JAMES-2096 Webadmin triggered migrations

### PR DESCRIPTION
**Pull request by chibenwa**
https://github.com/linagora/james-project/pull/906 rebased and fixuped.

### CODE REVIEW

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/Migration.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+package org.apache.james.mailbox.cassandra.mail.migration;
+
+public interface Migration {
+    /**
+     * Runs the migration
+     *
+     * @return Return true if fully migrated. Returns false otherwise.
+     */
+    boolean run();
```
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/Migration.java]: returning a boolean from run is not obvious (Runnable for example return void). If we intend to return some meaning, maybe an enum would be better?

File: src/site/markdown/server/manage-webadmin.md [hunk 2]
```diff
@@ -331,4 +331,97 @@ Response codes:
...
+## Cassandra Schema upgrades
+
+Cassandra upgrades implies the creation of a new table. Thus restarting James is needed, as new tables are created on restart.
+
+Once done, we ship code that tries to read from new tables, and if not possible backs up to old tables. You can thus safely run
+without running additional migrations.
+
+On the fly migration can be enabled. However, one might want to force the migration in a controlled fashion, and update
+automatically current schema version use (assess in the database old versions is no more used, as the corresponding tables are empty).
+Note that the this process is safe: we ensure the service is not running concurrently on this James instance, that it does not bump
```
Comment (rouazana) [inline src/site/markdown/server/manage-webadmin.md]: that the this?

File: src/site/xdoc/server/config-cassandra.xml [hunk 3]
```diff
@@ -126,6 +126,22 @@
...
 
+  <section name="Cassandra migration process">
+    <p>Cassandra upgrades implies the creation of a new table. Thus restarting James is needed, as new tables are created on restart.</p>
+
+    <p>Once done, we ship code that tries to read from new tables, and if not possible backs up to old tables. You can thus safely run
+    without running additional migrations.</p>
+
+    <p>On the fly migration can be enabled. However, one might want to force the migration in a controlled fashion, and update
+    automatically current schema version use (assess in the database old versions is no more used, as the corresponding tables are empty).
+    Note that the this process is safe: we ensure the service is not running concurrently on this James instance, that it does not bump
```
Comment (rouazana) [inline src/site/xdoc/server/config-cassandra.xml]: idem?

File: src/site/xdoc/server/config-spring-jpa-postgres.xml [hunk 4]
```diff
@@ -42,7 +42,7 @@
 
         <p>Example to install it on Debian:</p>
         <source>
-          sudo apt-get update && sudo apt-get install postgresql-9.4 postgresql-client-9.4
+          sudo apt-get update &amp;&amp; sudo apt-get install postgresql-9.4 postgresql-client-9.4
```
Comment (rouazana) [inline src/site/xdoc/server/config-spring-jpa-postgres.xml]: I don't understand you comment about rebuilding the website. You can keep it in this PR and just put it in a standalone commit, no?

Comment (aduprat) [inline src/site/xdoc/server/config-spring-jpa-postgres.xml]: This is already merged.

Comment (chibenwa) [inline src/site/xdoc/server/config-spring-jpa-postgres.xml]: Removed

**Comment by chibenwa** [comment]
test this please
